File: Yow.CoD.Finance.Domain/Contracts/LoanOverPaidEvent.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class LoanOverPaidEvent : Event
    {
        public LoanOverPaidEvent(DateTimeOffset transactionDate, decimal amount)
        {
            TransactionDate = transactionDate;
            Amount = amount;
        }
        public DateTimeOffset TransactionDate { get; }
        public decimal Amount { get; }
    }
}

File: java/domain/src/main/java/com/leecampbell/cod/domain/contracts/LoanOverPaidEvent.java
package com.leecampbell.cod.domain.contracts;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public final class LoanOverPaidEvent extends DomainEvent{
    
    private final OffsetDateTime transactionDateTime;
    private final BigDecimal amount;

    public LoanOverPaidEvent(OffsetDateTime transactionDateTime, BigDecimal amount) {
        this.transactionDateTime = transactionDateTime;
        this.amount = amount;
    }

    public OffsetDateTime getTransactionDateTime() {
        return this.transactionDateTime;
    }

    public BigDecimal getAmount() {
        return this.amount;
    }
}